Sources/Parsing/ParsingError.swift

```diff
@@ -458,6 +460,17 @@ extension ParsingError.Context {
       }
       return lhsInput.endIndex > rhsInput.endIndex
 
+    case let (lhsInput as Slice<[Substring]>, rhsInput as Substring):
+      return lhsInput.first.map {
+        $0.base != rhsInput.base
+        ? false
+        : $0.startIndex > rhsInput.startIndex
+      }
+      ?? false
+
+    case (is Substring, is Slice<[Substring]>):
+      return !(rhs > lhs)
```

Comment:
This properly sorts errors that thread between array slices of substrings and the underlying substrings.